src/main/java/gregtech/common/tools/ToolHardHammer.java

```diff
@@ -15,10 +15,16 @@
 import net.minecraft.world.World;
 
 import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 public class ToolHardHammer extends ToolBase {
 
+    private static final Set<String> HAMMER_TOOL_CLASSES = new HashSet<String>() {{
+        add("hammer"); add("pickaxe");
+    }};
```

Comment:
Probably better to use `ImmutableSet.of("hammer", "pickaxe")` or similar here. If the Set itself is mutable then a caller could modify it, and the double-brace approach will also create an anonymous inner class which is probably not desirable.

Reply:
I also think these should be immutable from our side to keep them safe. Regarding double braces they have to be addressed too.